{{ cookiecutter.project_slug }}/{{ cookiecutter.project_slug }}/users/views.py

```diff
@@ -45,6 +45,11 @@ class UserViewSet(DjoserUserViewSet):
     search_fields = ["email", "last_name", "first_name"]
 
     def perform_update(self, serializer):
+        is_same_user = (self.get_object() == self.request.user)
+
+        if is_same_user or self.get_object().role == "ADMIN" or self.request.user.role == "USER" or self.request.user.role == "EDITOR":
```

Comment:
This if statement is difficult to understand or follow. Can we break the logic up into one or two well-named functions so that the code is self-documenting? For example, it could look something like:

```
if user_is_self(...) or user_is_not_admin(...)
  make_role_read_only(...)
```

Reply:
@coreyshuman Are additional functions necessary? How about adding some additional variable names. So, developers won't need to know what self.get_object() and self.request.user are doing.

Reply:
@coreyshuman Okay. The code has been updated. I added a function and added two more variables.